### PR TITLE
Handle implicit flag dependencies in `mender setup` command

### PR DESCRIPTION
**Pull request by alfrunes**
changelog: none

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

### CODE REVIEW

File: cli/cli.go [hunk 1]
```diff
@@ -448,12 +450,29 @@ func (runOptions *runOptionsType) handleCLIOptions(ctx *cli.Context) error {
...
+		} else if ctx.IsSet("server-ip") {
+			ctx.Set("demo", "true")
+		}
+		ctx.Set("mender-professional", "false")
+	}
+
+	if ctx.IsSet("update-poll") || ctx.IsSet("inventory-poll") ||
+		ctx.IsSet("retry-poll") {
+		ctx.Set("demo", "false")
+	}
```
Comment (lluiscampos) [inline cli/cli.go]: :+1:

File: cli/cli.go [hunk 2]
```diff
@@ -448,12 +450,29 @@ func (runOptions *runOptionsType) handleCLIOptions(ctx *cli.Context) error {
...
+	// Check implicit flag dependencies
+	if ctx.IsSet("server-url") || ctx.IsSet("server-ip") {
+		if ctx.IsSet("server-url") && ctx.IsSet("server-ip") {
+			return errors.Errorf(errMsgConflictingArgumentsF,
+				"server-url", "server-ip")
+		} else if ctx.IsSet("server-ip") {
+			ctx.Set("demo", "true")
+		}
+		ctx.Set("mender-professional", "false")
+	}
```
Comment (lluiscampos) [inline cli/cli.go]: I still have this strange case. Setting `server-ip` and any (or all) of the interval flags (i.e. --update-poll 123) will then prompt me for serverURL.

Comment (alfrunes) [reply cli/cli.go]: Yeah, I can see the changes due to setting --update-poll takes precedence over `--server-ip` what is the expected behavior in this case? Should I raise a conflicting flags error? should we treat `--server-ip` as `--server-url`?

**Comment by alfrunes** [comment]
@lluiscampos; I've updated the commit to handle demo and polling-interval flags.

**Pull request merged**